File: week1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars = 0;

    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars <= 0);

    int cents = round(dollars * 100);
    int coins = 0;
    int QUARTERS = 25;
    int DIMES = 10;
    int NICKELS = 5;
    // int PENNIES = 1;

    if (cents > 0.25)
    {
        coins = coins + round(cents / QUARTERS);
        cents = cents % QUARTERS;
    }

    if (cents > 0.10)
    {
        coins = coins + round(cents / DIMES);
        cents = cents % DIMES;
    }

    if (cents > 0.05)
    {
        coins = coins + round(cents / NICKELS);
        cents = cents % NICKELS;
    }

    coins = coins + round(cents);
    printf("%i\n", coins);
}

File: pset6/cash/cash.py
from cs50 import get_float

change = get_float("Change Owed: ")
# If the user fails to provide a non-negative value, your program should re-prompt the user for a valid amount again and again until the user complies.
while True:
    if change > 0:
        break

PENNY = 1
NICKEL = 5
DIME = 10
QUARTER = 25
cents = change * 100
num_coins = 0

if cents >= 25:
    num_coins += cents // 25
    cents = cents % 25
    print(f'25: cents={cents} coins={num_coins}')

if cents >= 10:
    num_coins += cents // 10
    cents = cents % 10
    print(f'10: cents={cents} coins={num_coins}')

if cents >= 5:
    num_coins += cents // 5
    cents = cents % 5
    print(f'5: cents={cents} coins={num_coins}')

num_coins += cents
print(int(num_coins))